folium/map.py

```diff
@@ -205,14 +210,53 @@ def __init__(self, color='blue', icon_color='white', icon='info-sign',
         self.prefix = prefix
 
 
-class Marker(MacroElement):
+class BaseMarker(MacroElement):
+    """Base class for marker type objects with single or multiple coordinates.
+
+    Not for direct consumption.
+
+    Note that self.location can be either a single coordinate pair or
+    a list of coordinate pairs.
+
+    """
+    _template = None
+
+    def __init__(self, location=None, locations=None,
```

Comment:
While I don't like the possible confusion that `location` vs `locations`, this is better than the current state of things. (And I believe we are not encouraging users to use the `BaseMarker`.)

Reply:
You're right, the possible confusion is not okay. I'll try something to split it off better.

Reply:
I made a new `BaseMultiLocation` class for the vector layers. It doesn't inherit from Marker, so no confusion about `self.location` or `self.locations`.